### PR TITLE
Fixes communicator newsfeed app.

### PR DESCRIPTION
**Pull request by Atermonera**
Fixes non-terminal while loop in get_latest_news(), helper.dm
Newscaster posts now have a world.time-esque timestamp number, so they can be (almost) uniquely identified

[Tested](https://puu.sh/Fcwj6/f3cf2edd08.mp4) (.mp4)

"Latest news" posts are now sorted, in a sensible manner that most certainly does NOT use a while loop that doesn't terminate, murdering the entire server the moment a fourth news post is made from ANY system. And the post timestamps are from the first function I found in time.dm that seemed like a good fit, because there's so many of them at this point it's basically a matter of "I choose... _this_ one."

And since the fix that caused the server-crash was reverted, this isn't really a high priority, and speaking of which, I actually need to update this branch because I pulled from master before the revert so I could verify the issue.

### CODE REVIEW

File: code/game/objects/items/devices/communicator/UI.dm [hunk 1]
```diff
@@ -280,6 +282,10 @@
 		var/obj/O = cartridge.internal_devices[text2num(href_list["toggle_device"])]
 		cartridge.active_devices ^= list(O) // Exclusive or, will toggle its presence
 
+	if(href_list["newsfeed"])
+		world << "newsfeed [href_list["newsfeed"]]"
```
Comment (Leshana) [inline code/game/objects/items/devices/communicator/UI.dm]: You probably did not intend to keep this.

**Comment by TheFurryFeline** [comment]
Can we also see about getting #6531 fixed in some way?

**Comment by Atermonera** [comment]
That's beyond the scope of this PR, which is specifically to fix the runtime originally fixed by #6706, and to fix the server-crash issue revealed by that fix and caused by a non-terminal while() loop due to fucky list semantics. I haven't ever looked at that side of communicator functions, so I can't make any binding promises.

**Comment by TheFurryFeline** [comment]
> That's beyond the scope of this PR, which is specifically to fix the runtime originally fixed by #6706, and to fix the server-crash issue revealed by that fix and caused by a non-terminal while() loop due to fucky list semantics. I haven't ever looked at that side of communicator functions, so I can't make any binding promises.

Fair nuff. Just thought I'd bring it up since it's one of the parts of the communicator's functions. And you were here working on it.

**Pull request merged**